shared/lib_utility_rate_equations.cpp

```diff
@@ -173,65 +173,102 @@ void rate_data::init(int num_rec_yearly) {
     dc_hourly_peak = std::vector<ssc_number_t>(m_num_rec_yearly, 0);
 }
 
+bool rate_data::check_for_kwh_per_kw_rate(int units) {
+    return (units == 1) || (units == 3);
+}
+
 void rate_data::init_energy_rates(bool gen_only) {
 	// calculate the monthly net energy per tier and period based on units
 	for (int m = 0; m < (int)m_month.size(); m++)
 	{
 		// check for kWh/kW
-        size_t start_tier = 0;
-        size_t end_tier = (int)m_month[m].ec_tou_ub.ncols() - 1;
         size_t num_periods = (int)m_month[m].ec_tou_ub.nrows();
-        size_t num_tiers = end_tier - start_tier + 1;
+        size_t num_tiers = (int)m_month[m].ec_tou_ub.ncols(); // Likely to be overwritten later
```

Comment:
Is this finalized? Have the TOU upper bounds stored in columns or rows now?

Reply:
Yes, the convention in rate_data::setup_energy_rates is that periods are rows and tiers are columns. I'll add some comments in the header file to reflect this.